pulp_puppet_plugins/pulp_puppet/plugins/migrations/0005_puppet_module_name_change.py

```diff
@@ -0,0 +1,50 @@
+import logging
+
+from mongoengine import NotUniqueError
+
+from pulp.server.controllers import repository as repo_controller
+from pulp.server.db import model
+
+from pulp_puppet.plugins.db.models import Module
+
+_log = logging.getLogger('pulp')
+
+
+def migrate(*args, **kwargs):
+    """
+    For each puppet module check and if needed update module name format.
+    """
+    modules = Module.objects(__raw__={'name': {'$regex': '/|-'}})
```

Comment:
The `__raw__` query should be replaced with one of the [regex based searching methods](http://docs.mongoengine.org/guide/querying.html#string-queries). I think the one you want is `contains`.